src/daemon/Display.cpp

```diff
@@ -121,9 +121,21 @@ namespace SDDM {
             // set flags
             m_started = true;
 
+            // determine session type
+            const QString &autologinSession = mainConfig.Autologin.Session.get();
+            Session session;
+            if (findSessionEntry(mainConfig.XDisplay.SessionDir.get(), autologinSession)) {
+                session.setTo(Session::X11Session, autologinSession);
+            } else if (findSessionEntry(mainConfig.WaylandDisplay.SessionDir.get(), autologinSession)) {
+                session.setTo(Session::WaylandSession, autologinSession);
+            } else {
+                qCritical() << "Unable to find autologin session entry" << autologinSession;
+                return;
```

Comment:
in that case the Session will be invalid and we won't have an actual command to execute (Exec line was not read). maybe i can just emit loginFailed() before return